python/pyspark/errors/exceptions/connect.py

```diff
@@ -188,3 +188,10 @@ class SparkUpgradeException(SparkConnectGrpcException, BaseSparkUpgradeException
     """
     Exception thrown because of Spark upgrade from Spark Connect.
     """
+
+
+class SparkConnectRetryException(SparkConnectException):
+    """
+    An exception that can be thrown upstream when inside retry and which will be retryable
+    regardless of policy.
+    """
```

Comment:
nit: if this is an internal exception that is being thrown only inside Retrying, and not an user facing exception, should this be placed somewhere next to the AttemptManager and Retrying impl?

Reply:
Actually yeah, fair enough.